src/device-registry/bin/store-readings-job.js

```diff
@@ -52,9 +52,16 @@ const fetchAndStoreDataIntoReadingsModel = async () => {
             async (bail) => {
               try {
                 // logObject("document", doc);
-                const res = await ReadingModel("airqo").updateOne(doc, doc, {
-                  upsert: true,
-                });
+                const filter = { site_id: doc.site_id, time: doc.time };
```

Comment:
Explicitly stating the filter we shall use for the update operations. Previous version was using all the fields to filter which also included the _id field from the Events collection!